lms/templates/admin/base_site.html

```diff
@@ -20,3 +20,78 @@ <h1 id="site-name"><a href="{% url 'admin:index' %}">{{ site_header|default:_('D
+    <form id='survey_report_form' method="POST" action="/survey_report/generate_report" style="margin: 0; padding: 0;">
+        {% csrf_token %}
+        <button type="submit" style="background-color: #377D4D; color: var(--button-fg); border: none; border-radius: 4px; padding: 10px 20px; cursor: pointer;">Send Report</button>
+    </form>
+</div>
+</div>
+<div id="thankYouMessage" style="display: none; background-color: var(--darkened-bg); padding: 20px 40px; margin-bottom: 30px;box-shadow: rgb(0 0 0 / 18%) 0px 3px 5px;">
+    <div style="display: flex; align-items: center;">
+        <svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 24 24">
+            <g fill="#377D4D"><path d="M22 12c0 5.523-4.477 10-10 10S2 17.523 2 12S6.477 2 12 2s10 4.477 10 10Z"></path>
+                <path d="M16.03 8.97a.75.75 0 0 1 0 1.06l-5 5a.75.75 0 0 1-1.06 0l-2-2a.75.75 0 1 1 1.06-1.06l1.47 1.47l2.235-2.236L14.97 8.97a.75.75 0 0 1 1.06 0Z" fill="#FFF"></path>
+            </g>
+        </svg>   
+        <span style="font-size: 16px; margin-left: 15px;">Thank you for your collaboration and support! Your contribution is greatly appreciated and will help us continue to improve.</span>
+    </div>
+</div>
+{% endif %}
+
+    <!-- The original content of the block -->
+    <script>
+   $(document).ready(function(){
+    $('#dismissButton').click(function() {
+                $('#originalContent').slideUp('slow', function() {
+                    // If you want to do something after the slide-up, do it here.
+                    // For example, you can hide the entire div:
+                    // $(this).hide();
+                });
+            });
+    // When the form is submitted
+    $("#survey_report_form").submit(function(event){
+        event.preventDefault();  // Prevent the form from submitting traditionally
+        
+        // Make the AJAX request
+        $.ajax({
+            url: $(this).attr("action"),
+            type: $(this).attr("method"),
+            data: $(this).serialize(),
+            success: function(response){
+                // Hide the original content block
+                $("#originalContent").slideUp(400, function() {
+                    //$(this).css('display', 'none');
+                    // Show the thank-you message block with slide down effect
+                    $("#thankYouMessage").slideDown(400, function() {
+                        // Wait for 3 seconds (3000 milliseconds) and then slide up the thank-you message
+                        setTimeout(function() {
+                            $("#thankYouMessage").slideUp(400);
+                        }, 3000);
+                    });
+                });
+            },
+            error: function(error){
+                // Handle any errors
+                console.error("Error sending report:", error);
+            }
+        });
+    });
+});
+</script>
```

Comment:
Can this be a template fragment that lives in the survey report app, so that we keep the top level template shorter and easier to understand?

Reply:
This fragment of code moved up to the survey report folders to make the code cleaner